src/Signer/DataSigner.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the ACME PHP library.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Ssl\Signer;
+
+use AcmePhp\Ssl\Exception\DataSigningException;
+use AcmePhp\Ssl\PrivateKey;
+
+/**
+ * Provide tools to sign data using a private key.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class DataSigner
+{
+    /**
+     * Generate a signature of the given data using a private key and an algorithm.
+     *
+     * @param string     $data
+     * @param PrivateKey $privateKey
+     * @param int        $algorithm
+     *
+     * @return string
+     */
+    public function signData($data, PrivateKey $privateKey, $algorithm = OPENSSL_ALGO_SHA1)
```

Comment:
Actually you're right, sorry. Google and some others seems to deprecate usage of sha1 in favor of sha256 (https://security.googleblog.com/2014/09/gradually-sunsetting-sha-1.html), I'll test to get a certificate with a sha256 signature and if it works properly, I'll change it here.